### ISSUE TITLE
The alpha/beta version should attempt to re-run the CI after a failure to ensure NVDA updates are available.

### ISSUE THREAD

**Issue by hwf1324** [nvaccess/nvda#20061]
### What is the current state of the codebase?

It seems that no relevant measures have been taken.

### Why are changes required?

Our CI frequently fails due to flaky system tests, which is a hassle for alpha/beta users who want to test new changes in a timely manner.

### What technical changes are required?

System test stability should be guaranteed whenever possible.

A retry mechanism should be implemented for CI failures on `master/beta/rc/try-*` branches, such as automatically retrying up to 3 times.

### Are the proposed technical changes API breaking?

No

### Are there potential risks or issues with the proposed implementation?

May consume more CI resources?
